MapboxDirections/MBRouteLeg.swift

```diff
@@ -98,6 +130,9 @@ open class RouteLeg: NSObject, NSSecureCoding {
      */
     open let steps: [RouteStep]
     
+    
+    open let annotation: [AnnotationType:[Any]]?
```

Comment:
Annotation is a poor name for this information for the following reasons:

* It's a collection of data, so a plural word is preferable.
* The Mapbox iOS and macOS SDKs already have a concept of annotations (for consistency with MapKit) that's completely different: markers and overlays on a map. Google Maps calls them markers, but developers are required to use the Mapbox SDKs with this library anyways.
* The term doesn't say anything about the relationship between the route leg and the information indicated by the property.

Some suggestions:

* nodeAttributes (so far only for the nodeIdentifiers key) and segmentAttributes (for the other keys)
* attributes
* userInfo – Although this may sound strange, it's consistent with Cocoa APIs like NSError and NSNotification.